src/server/tiering/common.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "util/fibers/synchronization.h"
+
+namespace dfly::tiering {
+
+// Interface for simple fiber-blocking disk storage.
+class Storage {
+ public:
+  virtual std::string Read(std::string_view key) = 0;
+  virtual void Write(std::string_view key, std::string value) = 0;
+  virtual void Delete(std::string_view key) = 0;
+};
```

Comment:
I understand that just "key" is not enough, the value stored the file offset/ blobl size. I will update this once I'll exetend the interface